File: src/Module/Group/GroupEntity.java
package Module.Group;

import Module.Employee.EmployeeEntity;
import Module.Employee.EmployeeModel;
import Module.Team.TeamEntity;
import Module.Team.TeamModel;
import Module.Ticket.TicketEntity;
import Module.Ticket.TicketModel;

import java.io.Serializable;
import java.util.Collection;
import java.util.List;
import java.util.stream.Collectors;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
public class GroupEntity implements Serializable {
    public int id;
    public String name;
    public int managerEmloyeeId;
    public EmployeeEntity managerEmployeeEntity;
    public List<EmployeeEntity> employeeEntities;
    public List<TeamEntity> teamEntities;
    public List<TicketEntity> ticketEntities;


    public GroupEntity() {
    }

    public GroupEntity(GroupModel GroupModel, Object... objects) {
        this.id = GroupModel.getId();
        this.name = GroupModel.getName();
        this.managerEmloyeeId = GroupModel.getManagerEmloyeeId();
        for (Object object : objects) {
            if (object instanceof EmployeeModel) {
                this.managerEmployeeEntity = new EmployeeEntity((EmployeeModel) object);
            } else if (object instanceof Collection) {
                for (Object o : (Collection<Object>) object) {
                    if (o instanceof EmployeeModel) {
                        this.employeeEntities = ((Collection<EmployeeModel>) object).parallelStream().map(EmployeeEntity::new).collect(Collectors.toList());
                        break;
                    } else if (o instanceof TeamModel) {
                        this.teamEntities = ((Collection<TeamModel>) object).parallelStream().map(TeamEntity::new).collect(Collectors.toList());
                        break;
                    } else if (o instanceof TicketModel) {
                        this.ticketEntities = ((Collection<TicketModel>) object).parallelStream().map(TicketEntity::new).collect(Collectors.toList());
                        break;
                    }
                }
            }
        }
    }

    public GroupModel toModel() {
        GroupModel GroupModel = new GroupModel();
        GroupModel.setId(id);
        GroupModel.setName(name);
        GroupModel.setManagerEmloyeeId(managerEmloyeeId);
        return GroupModel;
    }
}

File: web/Angular4/src/app/Modules/Group/Group.Entity.ts
import {EmployeeEntity} from "../Employee/Employee.Entity";
import {TeamEntity} from "../Team/Team.Entity";
import {TicketEntity} from "../Ticket/Ticket.Entity";

export class GroupEntity {

    id: number;
    name: string;
    managerEmloyeeId: number;
    managerEmployeeEntity : EmployeeEntity;
    employeeEntities: EmployeeEntity[];
    teamEntities: TeamEntity[];
    ticketEntities: TicketEntity[];

    IsEdit: boolean;
    IsActive: boolean = false;
    IsSelected: boolean = false;
    
    constructor(data: any = null) {
        if (data == null) {
            this.id = null;
            this.name = null;
            this.managerEmloyeeId = null;
            this.managerEmployeeEntity = new EmployeeEntity();
            this.employeeEntities = [];
            this.teamEntities = [];
            this.ticketEntities = [];
        } else {
            this.id = data.id;
            this.name = data.name;
            this.managerEmloyeeId = data.managerEmloyeeId;
            if (data.employeeEntities != null) {
                this.employeeEntities = [];
                for (let item of data.employeeEntities) {
                    this.employeeEntities.push(new EmployeeEntity(item));
                }
            }
            if (data.teamEntities != null) {
                this.teamEntities = [];
                for (let item of data.teamEntities) {
                    this.teamEntities.push(new TeamEntity(item));
                }
            }
            if (data.ticketEntities != null) {
                this.ticketEntities = [];
                for (let item of data.ticketEntities) {
                    this.ticketEntities.push(new TicketEntity(item));
                }
            }
            if (this.managerEmployeeEntity == null) this.managerEmployeeEntity = new EmployeeEntity();
            if (this.teamEntities == null) this.teamEntities = [];
            if (this.employeeEntities == null) this.employeeEntities = [];
            if (this.ticketEntities == null) this.ticketEntities = [];
        }
        this.IsEdit = false;
    }
}